src/Tokenizer.js

```diff
@@ -101,15 +103,16 @@ export function token (type) {
 
 /**
  * @param {string} value
- * @return {any[]}
+ * @return {[]}
  */
 export function alloc (value) {
 	return line = column = 1, length = strlen(characters = value), position = 0, []
 }
 
 /**
- * @param {any} value
- * @return {any}
+ * @typedef {any} T
+ * @param {T} value
+ * @return {T}
```

Comment:
I'm trying to show that what you give it is what you get back, as `T`

Reply:
this could use `@template T`

Reply:
I think that's a limitation of the way the code is written - If this package *had any documentation* it'd be good for it to go in there. I agree that it's confusing, but I think a lot of this codebase is. A lot of functions have side effects on shared global data, or are aliases to other functions, or change data types randomly in order to have "terse" code. It feels like there's too much to fix in terms of type safety.

I'm making another comment in this thread in a sec, basically saying I'd be happy for you to take over since I imagine this PR will stall.